src/client.lisp

```diff
@@ -0,0 +1,95 @@
+;;;; client.lisp
+;;;;
+;;;; Author: Eric Peterson
+;;;;
+;;;; Lisp mimic of the python JSON RPC Client.
+
+
+(in-package #:rpcq)
+
+(defstruct rpc-client
+  "Holds the data for an (active) RPCQ client connection."
+  socket
+  timeout)
+
+(define-condition RPC-error (simple-error)
+  ((error-string :initarg :string :reader RPC-error-string)
+   (request-id :initarg :id :reader RPC-error-request-id))
+  (:report (lambda (condition stream)
+             (format stream "RPC request ~a resulted in error:~%~a~&"
+                     (RPC-error-request-id condition)
+                     (RPC-error-string condition)))))
+
+(define-condition RPC-protocol-error (simple-error)
+  ((object :initarg :object :reader RPC-protocol-error-object)
+   (id :initarg :id :reader RPC-protocol-error-id))
+  (:report (lambda (condition stream)
+             (format stream "RPC request ~a resulted in bad reply:~%~a~&"
+                     (RPC-protocol-error-id condition)
+                     (RPC-protocol-error-object condition)))))
+
+
+(defun rpc-call (client call &rest args)
+  "Makes a synchronous RPC client call over a client connection.  Returns the result of the call directly."
+  (let* ((uuid (unicly:uuid-princ-to-string (unicly:make-v4-uuid)))
+         (request (make-instance '|RPCRequest|
+                                 :|id| uuid
+                                 :|params| (alexandria:plist-hash-table args :test #'equal)
+                                 :|method| (sanitize-name call)))
+         (payload (serialize request))
+         (socket (rpc-client-socket client)))
+    (cffi:with-foreign-object (foreign-payload :unsigned-char (length payload))
+      (dotimes (j (length payload))
+        (setf (cffi:mem-aref foreign-payload ':unsigned-char j)
+              (aref payload j)))
+      (pzmq:send socket foreign-payload :len (length payload)))
+    (labels
```

Comment:
Can you add a comment about the fact that this is tail recursive? `XXX: ...`

Would help the unsuspecting non-SBCL user.

Reply:
Non-SBCL users will also be upset by the occasional mention of `sb-ext:timeout`.